src/go/rpk/pkg/cli/transform/logs.go

```diff
@@ -0,0 +1,470 @@
+					}
+					maxOffset = min(maxOffset, untilOffset.Offset)
+				}
+			}
+			zap.L().Sugar().Infof("reading logs topic with bounds [%v, %v] on partition %d", startOffset, maxOffset, partition)
+			if startOffset > maxOffset {
+				return
+			}
+
+			consumeStart := map[string]map[int32]kgo.Offset{
+				topic.Topic: {
+					partition: kgo.NewOffset().At(startOffset),
+				},
+			}
+			client, err := kafka.NewFranzClient(fs, p, kgo.ConsumePartitions(consumeStart))
+			out.MaybeDie(err, "unable to create client: %v", err)
+
+			sigs := make(chan os.Signal, 2)
+			signal.Notify(sigs, os.Interrupt, syscall.SIGTERM, syscall.SIGHUP)
+			donePrinting := make(chan struct{})
+			var output io.Writer = os.Stdout
+			// kgo only supports consuming in a single direction per client, the simplest way to get
+			// the last N elements is to just read them all, and buffer the last N
+			if tail > 0 {
+				tw := &tailWriter{w: output, limit: tail}
+				defer tw.Close()
+				output = tw
+			}
+			go func() {
+				defer close(donePrinting)
+				printLogs(
+					cmd.Context(),
+					&clientLogSource{client},
+					logsQuery{
+						transformName: transformName,
+						maxOffset:     maxOffset,
+						limit:         head,
+					},
+					output,
+					f,
+				)
+			}()
+
+			select {
+			case <-sigs:
+			case <-donePrinting:
+			}
+
+			doneClose := make(chan struct{})
+			go func() {
+				defer close(doneClose)
+				client.Close()
+			}()
+
+			select {
+			case <-sigs:
+			case <-doneClose:
+			}
+		},
+	}
+	cmd.Flags().BoolVarP(&follow, "follow", "f", false, "Specify if the logs should be streamed")
+	cmd.Flags().IntVar(&head, "head", 0, "The number of log entries to fetch from the start")
+	cmd.Flags().IntVar(&tail, "tail", 0, "The number of log entries to fetch from the end")
+	cmd.MarkFlagsMutuallyExclusive("head", "tail")
+	// Tail and follow cannot be used together, if you're streaming forever it doesn't make since to get the "last" N logs.
+	cmd.MarkFlagsMutuallyExclusive("tail", "follow")
+	cmd.Flags().Var(&since, "since", "Start reading logs after this time")
+	cmd.Flags().Var(&until, "until", "Read logs up unto this time")
+	p.InstallFormatFlag(cmd)
+	return cmd
+}
+
+func computeLogPartition(name string, tp kadm.TopicPartitions) int32 {
+	// Use the murmur2 hash function to compute the partition
+	p := kgo.StickyKeyPartitioner(nil)
+	r := kgo.KeyStringRecord(name, "")
+	return int32(p.ForTopic(tp.Topic).Partition(r, len(tp.Partitions)))
+}
+
+type timequery struct {
+	time time.Time
+}
+
+func (tq *timequery) Set(s string) (err error) {
+	tq.time, err = parseTimeQuery(s, time.Now())
+	return
+}
+
+func (*timequery) Type() string {
+	return "timequery"
+}
+
+func (tq *timequery) String() string {
+	// Return empty string for defaults
+	if tq.time.Equal(minTime) || tq.time.Equal(maxTime) {
+		return ""
+	}
+	return tq.time.String()
+}
+
+func parseTimeQuery(s string, now time.Time) (time.Time, error) {
```

Comment:
I think this is the same one used in `consume`, if so, I wonder if we can abstract that to `pkg/out/in.go` wdyt?

Reply:
It's very similar, but different in that it doesn't support the range syntax with colons like consume. I'm also going to add a `now` shorthand so you get get new logs with `--since=now`. I don't know if it still makes since to merge these unfortunately.

Reply:
Ohh, got it... It's fine :+1: thanks for checking it !